src/WebUI/Components/StatefulSetListingComponent.tsx

```diff
@@ -0,0 +1,125 @@
+import { IStatusProps, Status, Statuses, StatusSize } from "azure-devops-ui/Status";
+import { ColumnActionsMode } from "office-ui-fabric-react/lib/DetailsList";
+import * as React from "react";
+import * as Resources from "../Resources";
+import { ListComponent } from "./ListComponent";
+import { IVssComponentProperties } from "../Types";
+import { Ago } from "azure-devops-ui/Ago";
+import { Utils } from "../Utils";
+import "./StatefulSetListingComponent.scss";
+
+const setNameKey = "statefulset-name-key";
+const imageKey = "statefulset-image-key";
+const podsKey = "statefulset-pods-key";
+const ageKey = "statefulset-age-key";
+
+export interface IDaemonSetComponentProperties extends IVssComponentProperties {
+    statefulSetList: V1StatefulSetList;
+    onItemInvoked?: (item?: any, index?: number, ev?: Event) => void;
+}
+
+
+export class StatefulSetListingComponent extends BaseComponent<IDaemonSetComponentProperties, {}> {
+    public render(): React.ReactNode {
+        return (
+            <div>{
+
+                <ListComponent
+                    className={css("list-content", "top-padding", "depth-16" )}
+                    items={ this.props.statefulSetList.items || [] }
+                    columns={StatefulSetListingComponent._getColumns()}
+                    onRenderItemColumn={StatefulSetListingComponent._onRenderItemColumn}
+                />
+            }</div>
+        );
+    }
+
+    private static _getColumns(): IColumn[] {
+        let columns: IColumn[] = [];
+        const headerColumnClassName: string = "secondary-text";
+
+        columns.push({
+            key: setNameKey,
+            name: Resources.StatefulSetText,
+            fieldName: setNameKey,
+            minWidth: 250,
+            maxWidth: 250,
+            headerClassName: css(headerColumnClassName, "first-col-header"),
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+        columns.push({
+            key: imageKey,
+            name: Resources.ImageText,
+            fieldName: imageKey,
+            minWidth: 250,
+            maxWidth: 250,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+        columns.push({
+            key: podsKey,
+            name: Resources.PodsText,
+            fieldName: podsKey,
+            minWidth: 80,
+            maxWidth: 80,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+        columns.push({
+            key: ageKey,
+            name: Resources.AgeText,
+            fieldName: ageKey,
+            minWidth: 80,
+            maxWidth: 80,
+            headerClassName: headerColumnClassName,
+            columnActionsMode: ColumnActionsMode.disabled
+        });
+
+
+        return columns;
+    }
+    private static _onRenderItemColumn(statefulSet?: V1StatefulSet, index?: number, column?: IColumn): React.ReactNode {
```

Comment:
common for many other components, can we have a new component for this structure?